src/main/java/appeng/core/sync/packets/PacketInventoryAction.java

```diff
@@ -22,54 +22,42 @@
 import java.io.IOException;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.tileentity.TileEntity;
 
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
+import cpw.mods.fml.relauncher.Side;
+
 import appeng.api.storage.data.IAEItemStack;
 import appeng.client.ClientHelper;
 import appeng.container.AEBaseContainer;
 import appeng.container.ContainerOpenContext;
 import appeng.container.implementations.ContainerCraftAmount;
 import appeng.core.sync.AppEngPacket;
+import appeng.core.sync.AppEngPacketHandler;
 import appeng.core.sync.GuiBridge;
-import appeng.core.sync.network.INetworkInfo;
 import appeng.helpers.InventoryAction;
 import appeng.util.Platform;
 import appeng.util.item.AEItemStack;
 
 
-public class PacketInventoryAction extends AppEngPacket
+public class PacketInventoryAction implements AppEngPacket, AppEngPacketHandler<PacketInventoryAction, AppEngPacket>
 {
 
-	public final InventoryAction action;
-	public final int slot;
-	public final long id;
-	public final IAEItemStack slotItem;
+	private InventoryAction action;
+	private int slot;
+	private long id;
+	private IAEItemStack slotItem;
 
 	// automatic.
-	public PacketInventoryAction( ByteBuf stream ) throws IOException
+	public PacketInventoryAction()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)